com.microsoft.mrtk.spatialmanipulation/BoundsControl/Visuals/SpatialManipulationReticle.cs

```diff
@@ -0,0 +1,128 @@
+//Copyright(c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.SpatialManipulation
+{
+    public class SpatialManipulationReticle : MonoBehaviour
+    {
+
+        [SerializeField]
+        [Tooltip("The type of the reticle visuals. Scale or Rotate.")]
+        private SpatialManipulationReticleType reticleType;
+
+        private Quaternion worldRotationCache;
+
+        /// <summary>
+        /// Called by once per frame by <see cref="MRTKRayReticleVisual"/> from its UpdateReticle.
+        /// Rotates the cursor reticle based on the hovered or selected handle's position relative to the box visuals. 
+        /// </summary>
+        public void RotateReticle(Vector3 reticleNormal, Transform hitTargetTransform)
+        {
+            // After hitting a handle, find the box that the handle belongs to
+            SqueezableBoxVisuals boxVisuals = hitTargetTransform.gameObject.GetComponentInParent<SqueezableBoxVisuals>(true);
```

Comment:
We should avoid querying for `GetComponentInParent<SqueezableBoxVisuals>` every frame.  Can you cache this somehow?